src/gui/qgscoordinatereferencesystemmodel.cpp

```diff
@@ -0,0 +1,751 @@
+        groupName = tr( "Geographic (3D)" );
+        groupIcon = QgsApplication::getThemeIcon( QStringLiteral( "/mIconProjectionEnabled.svg" ) );
+        break;
+
+      case Qgis::CrsType::Vertical:
+        groupName = tr( "Vertical" );
+        break;
+
+      case Qgis::CrsType::Projected:
+      case Qgis::CrsType::DerivedProjected:
+        groupName = tr( "Projected" );
+        groupIcon = QgsApplication::getThemeIcon( QStringLiteral( "/transformed.svg" ) );
+        break;
+
+      case Qgis::CrsType::Compound:
+        groupName = tr( "Compound" );
+        break;
+
+      case Qgis::CrsType::Temporal:
+        groupName = tr( "Temporal" );
+        break;
+
+      case Qgis::CrsType::Engineering:
+        groupName = tr( "Engineering" );
+        break;
+
+      case Qgis::CrsType::Bound:
+        groupName = tr( "Bound" );
+        break;
+
+      case Qgis::CrsType::Other:
+        groupName = tr( "Other" );
+        break;
+    }
+  }
+
+  if ( QgsCoordinateReferenceSystemModelGroupNode *group = parentNode->getChildGroupNode( groupId ) )
+  {
+    parentNode = group;
+  }
+  else
+  {
+    std::unique_ptr< QgsCoordinateReferenceSystemModelGroupNode > newGroup = std::make_unique< QgsCoordinateReferenceSystemModelGroupNode >( groupName, groupIcon, groupId );
+    parentNode->addChildNode( newGroup.get() );
+    parentNode = newGroup.release();
+  }
+
+  if ( ( record.authName != QLatin1String( "USER" ) && record.authName != QLatin1String( "CUSTOM" ) ) && ( record.type == Qgis::CrsType::Projected || record.type == Qgis::CrsType::DerivedProjected ) )
+  {
+    QString projectionName = QgsCoordinateReferenceSystemUtils::translateProjection( record.projectionAcronym );
+    if ( projectionName.isEmpty() )
+      projectionName = tr( "Other" );
+
+    if ( QgsCoordinateReferenceSystemModelGroupNode *group = parentNode->getChildGroupNode( record.projectionAcronym ) )
+    {
+      parentNode = group;
+    }
+    else
+    {
+      std::unique_ptr< QgsCoordinateReferenceSystemModelGroupNode > newGroup = std::make_unique< QgsCoordinateReferenceSystemModelGroupNode >( projectionName, QIcon(), record.projectionAcronym );
+      parentNode->addChildNode( newGroup.get() );
+      parentNode = newGroup.release();
+    }
+  }
+
+  parentNode->addChildNode( crsNode.get() );
+  return crsNode.release();
+}
+
+QModelIndex QgsCoordinateReferenceSystemModel::addCustomCrs( const QgsCoordinateReferenceSystem &crs )
+{
+  QgsCrsDbRecord userRecord;
+  userRecord.authName = QStringLiteral( "CUSTOM" );
+  userRecord.description = crs.description().isEmpty() ? tr( "Custom CRS" ) : crs.description();
+  userRecord.type = crs.type();
+
+  QgsCoordinateReferenceSystemModelGroupNode *group = mRootNode->getChildGroupNode( QStringLiteral( "CUSTOM" ) );
+  if ( !group )
+  {
+    std::unique_ptr< QgsCoordinateReferenceSystemModelGroupNode > newGroup = std::make_unique< QgsCoordinateReferenceSystemModelGroupNode >(
+          tr( "Custom Coordinate Systems" ),
```

Comment:
Is it a Custom Coordinate >>Reference<< System or a Custom Coordinate System? I'm asking as a few lines above, you're using "Custom C>>R<<S.